src/ldp/operations/GetOperationHandler.ts

```diff
@@ -26,7 +29,9 @@ export class GetOperationHandler extends OperationHandler {
   public async handle(input: Operation): Promise<ResponseDescription> {
     const body = await this.store.getRepresentation(input.target, input.preferences, input.conditions);
 
-    input.authorization?.addMetadata(body.metadata);
+    if (input.permissionSet) {
```

Comment:
Alternatively, can the `LdpHandler` _always_ collect the metadata (on line 96)?
And we just write `WAC-Allow` conditionally to the result?

Then we don't need to set `operation.permissionSet` either for now?

Reply:
The LDP handler already always collects permissions in the current implementation so we could also safely use an `!` operator here, or create a new `AuthorizedOperation` interface where the value is not optional. I think this was one of the suggestions when we added this some time ago but we instead opted for the optional parameter then.

On the other hand it's not really an issue to have this if-statement and it does provide more flexibility if we want to reuse these operation handlers somewhere else where there is no authorization, so I would keep it like this.

Reply:
> The LDP handler already always collects permissions in the current implementation

What I meant was: should the

```typescript
await this.metadataWriter.handleSafe({ metadata: body.metadata, permissionSet: input.permissionSet });
```

perhaps be performed unconditionally by the `LdpHandler`, instead of doing it in `GET` and `HEAD` handlers?
Such that this becomes part of the generic request flow? (The "unconditionally" does not refer to the `if` statement, but to to the fact that only `GET` and `HEAD` have it now.)

No strong opinion.

Reply:
Makes sense, but would also require us to pass the entire operation instead of just the permissionset since WAC-Allow should only be there for GET/HEAD requests, so going to leave it like this until we need more in the future.

Reply:
> since WAC-Allow should only be there for GET/HEAD requests

But that should be a WAC-Allow concern, no?

I'd expect the `if` to be at that point. I think it's quite some coupling to have this metadata conditionally in an operation, with the expected output that, down the line, `WAC-Allow` is only written out for certain methods. That knowledge should only be in `WAC-Allow` IMHO, such that, if that header ever changes, only that class needs updating.

Reply:
> 
> 
> > since WAC-Allow should only be there for GET/HEAD requests
> 
> But that should be a WAC-Allow concern, no?

Yea that's what I meant when I said the entire operation needs to be passed as input parameter then, since WAC-Allow needs to know which method was used.

Since I have the impression that you would prefer that solution I'm going to go with that then 😄

Reply:
At that point we should probably call it an `OperationMetadataCollector` though.

Reply:
> At that point we should probably call it an `OperationMetadataCollector` though.

I like that!

> the entire operation needs to be passed as input parameter then

Well, it could be part of that metadata… 😅

Could be a "read-only" flag. Or we could use the Safe/Idempotent from HTTP.

Or just the method directly, yeah.